conans/assets/templates/new_v2_autotools.py

```diff
@@ -0,0 +1,360 @@
+import textwrap
+
+conanfile_lib = textwrap.dedent("""
+    import os
+
+    from conan import ConanFile
+    from conan.tools.gnu import AutotoolsToolchain, Autotools
+    from conan.tools.gnu.layout import autotools_layout
+    from conan.tools.files import chdir
+
+
+    class {package_name}Conan(ConanFile):
+        name = "{name}"
+        version = "{version}"
+
+        # Optional metadata
+        license = "<Put the package license here>"
+        author = "<Put your name here> <And your email here>"
+        url = "<Package recipe repository url here, for issues about the package>"
+        description = "<Description of {package_name} here>"
+        topics = ("<Put some tag here>", "<here>", "<and here>")
+
+        # Binary configuration
+        settings = "os", "compiler", "build_type", "arch"
+        options = {{"shared": [True, False], "fPIC": [True, False]}}
+        default_options = {{"shared": False, "fPIC": True}}
+
+        exports_sources = "configure.ac", "Makefile.am", "src/*"
+
+        def config_options(self):
+            if self.settings.os == "Windows":
+                del self.options.fPIC
+
+        def layout(self):
+            autotools_layout(self)
+
+        def generate(self):
+            at_toolchain = AutotoolsToolchain(self)
+            at_toolchain.generate()
+
+        def build(self):
+            with chdir(self, self.source_folder):
+                self.run("aclocal")
+                self.run("autoconf")
+                self.run("automake --add-missing --foreign")
+            autotools = Autotools(self)
+            autotools.configure(self.source_folder)
+            autotools.make()
+
+        def package(self):
+            autotools = Autotools(self)
+            autotools.install()
+
+        def package_info(self):
+            self.cpp_info.libs = ["{name}"]
+    """)
+
+source_h = textwrap.dedent("""
```

Comment:
This files can be reused from other template maybe? At least in 2.0 I just import them from cmake template

Reply:
Yes, I'm making a second pass today and refactoring everything 👍